lxd/device/cdi/configure.go

```diff
@@ -291,14 +291,35 @@ func GenerateFromCDI(s *state.State, inst instance.Instance, cdiID ID, l logger.
 	configDevices := &ConfigDevices{UnixCharDevs: make([]map[string]string, 0), BindMounts: make([]map[string]string, 0)}
 
 	// 2. Process the specific device configuration
+	lookedUpDevs := make(map[string]struct{})
 	for _, device := range spec.Devices {
-		if device.Name == cdiID.Name {
-			mounts, err = applyContainerEdits(device.ContainerEdits, configDevices, hooks, mounts, l)
-			if err != nil {
-				return nil, nil, err
+		if cdiID.Name == "all" {
+			devToAdd := true
+			for _, devNode := range device.ContainerEdits.DeviceNodes {
```

Comment:
Could you add a comment here explaining why you are doing this part.

Reply:
fixed